pkg/cmd/server/origin/master_config.go

```diff
@@ -675,11 +675,20 @@ func newAdmissionChain(pluginNames []string, admissionConfigFilename string, plu
 				return nil, err
 			}
 
-			plugin, err = admissionregistry.OriginAdmissionPlugins.InitPlugin(pluginName, pluginConfigReader, admissionInitializer)
+			plugin = nil
+			enabled, pluginConfigReaderCopy, err := admissionregistry.IsAdmissionPluginEnabled(pluginName, pluginConfigReader)
```

Comment:
Why isn't this the upstream `admission.PluginEnabledFn` ?

Reply:
How would you avoid passing DefaultAdmissionConfig to admission plugins if you did that? Also by doing this here gives us 2 advantages as I understand:
1. we do not need to override upstream admission.PluginEnabledFn
2. We do not need to send disabled plugins all the way down as I said above.

Reply:
> We do not need to send disabled plugins all the way down as I said above.

What do you mean by "all the way down"?



> we do not need to override upstream admission.PluginEnabledFn

We created that for exactly this purpose.

Reply:
>What do you mean by "all the way down"?

I mean to avoid InitPlugin->getPlugin-> (splitstream, PluginEnabledFn) inside kubernetes.

> We created that for exactly this purpose.

I agree but how would you avoid DefaultAdmissionConfig passing to admission plugins if we used that. We need a way to process  DefaultAdmissionConfig and a plugin's own config differently.